File: app/src/main/java/com/jmalinen/blackjack/model/Card.kt
package com.jmalinen.blackjack.model

data class Card(
    val rank: Rank,
    val suit: Suit,
    val faceUp: Boolean = true
) {
    val display: String get() = "${rank.symbol}${suit.symbol}"

    fun flip(): Card = copy(faceUp = !faceUp)
}

File: ios/Blackjack/Blackjack/Model/Card.swift
struct Card: Equatable, Hashable {
    let rank: Rank
    let suit: Suit
    var faceUp: Bool = true

    var display: String { "\(rank.symbol)\(suit.symbol)" }

    func flip() -> Card {
        Card(rank: rank, suit: suit, faceUp: !faceUp)
    }
}
